### ISSUE TITLE
Expose DEFAULT_SETTINGS or have CustomThemeProvider part of "settings"

### ISSUE THREAD

**Issue by jimmy-guzman** [semoal/themeprovider-storybook#43]
**Is your feature request related to a problem? Please describe.**
With the introduction of `disableThemePreview`, what is mentioned in the README no longer works for `CustomThemeProvider` since it's now the [3rd param](https://github.com/semoal/themeprovider-storybook/blob/master/src/withThemesProvider.tsx#L19)
```jsx
import { SomeCustomImplementationOfThemeProvider, DEFAULT_SETTINGS } from "src/app/CustomThemeProvider.jsx"

addDecorator(
  withThemesProvider(themes, SomeCustomImplementationOfThemeProvider)
);
```

**Describe the solution you'd like**

* This is mitigated simply by updating the README to `withThemesProvider(themes, { disableThemePreview: false }, SomeCustomImplementationOfThemeProvider)` but it would be nice if [DEFAULT_SETTINGS](https://github.com/semoal/themeprovider-storybook/blob/master/src/withThemesProvider.tsx#L12) is exposed so we can do:
```jsx
import { SomeCustomImplementationOfThemeProvider, DEFAULT_SETTINGS } from "src/app/CustomThemeProvider.jsx"

addDecorator(
  withThemesProvider(themes, DEFAULT_SETTINGS, SomeCustomImplementationOfThemeProvider)
);
```

**Describe alternatives you've considered**
* Make `CustomThemeProvider` part of `ThemesProviderSettings` so we can do something like: (preferred but it's a breaking change which can be mitigated by supporting both options until the next planned major version)
```jsx
withThemesProvider(themes, { CustomThemeProvider: SomeCustomImplementationOfThemeProvider })
```

**Additional context**
I'm willing to work on either solution :)

**Comment by jimmy-guzman** [comment]
Also, any thoughts on changing the signature for `CustomThemeProvider` to a type has `children` as required since `React.ComponentType<{ theme: Theme }>` treats `children` as optional and it should be required? Right now to keep the correct signature on our custom theme providers we're having to do:
```tsx
ThemeProvider as React.ComponentType<{ theme: Theme }>
``` 

Liking this library so far 👍

**Comment by semoal** [comment]
Shiet, you're right Jimmy :/ forgot to made it compatible and broke the version a bit hehe..

I like your proposal of making settings adaptable to `CustomThemeProvider` also I'll add that if the 2nd param is a valid element use that for `CustomThemeProvider`.

I was thinking this week in refactor `react-json-view` to a more lightweight library (probably self-made) to reduce drastically the bundle-size, so probably i'll release a major in a few days.

Meanwhile, probably I'll spoil a bit the code for making it compatible for everyone. 

Let me create a merge request and if you have some free time you can review it =)

Thanks mate for using the library hope you found it useful

**Comment by jimmy-guzman** [comment]
Thanks @semoal for quickly replying. The changes look good. 👍

**Comment by semoal** [comment]
Released 1.7.1 mate, thanks for opening the issue
